pkg/controller/certificatemanager/controller.go

```diff
@@ -0,0 +1,308 @@
+		for i, t := range cr.Spec.ForProvider.Tags {
+			tags[i] = awsacmpca.Tag{Key: aws.String(t.Key), Value: aws.String(t.Value)}
+		}
+
+		currentTags, err := e.client.ListTagsRequest(&awsacmpca.ListTagsInput{
+			CertificateAuthorityArn: aws.String(cr.Status.AtProvider.CertificateAuthorityArn),
+		}).Send(ctx)
+
+		if err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, errListTagsFailed)
+		}
+
+		if len(tags) < len(currentTags.Tags) {
+			_, err := e.client.UntagCertificateAuthorityRequest(&awsacmpca.UntagCertificateAuthorityInput{
+				CertificateAuthorityArn: aws.String(cr.Status.AtProvider.CertificateAuthorityArn),
+				Tags:                    currentTags.Tags,
+			}).Send(ctx)
+			if err != nil {
+				return managed.ExternalUpdate{}, errors.Wrap(err, errRemoveTagsFailed)
+			}
+		}
+		_, err = e.client.TagCertificateAuthorityRequest(&awsacmpca.TagCertificateAuthorityInput{
+			CertificateAuthorityArn: aws.String(cr.Status.AtProvider.CertificateAuthorityArn),
+			Tags:                    tags,
+		}).Send(ctx)
+		if err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, errAddTagsFailed)
+		}
+	}
+
+	if cr.Spec.ForProvider.CertificateRenewalPermissionAllow != cr.Status.AtProvider.RenewalPermission {
+
+		cr.Status.AtProvider.RenewalPermission = cr.Spec.ForProvider.CertificateRenewalPermissionAllow
+
+		if cr.Spec.ForProvider.CertificateRenewalPermissionAllow {
+
+			_, err = e.client.CreatePermissionRequest(&awsacmpca.CreatePermissionInput{
+				Actions:                 []awsacmpca.ActionType{awsacmpca.ActionTypeIssueCertificate, awsacmpca.ActionTypeGetCertificate, awsacmpca.ActionTypeListPermissions},
+				CertificateAuthorityArn: aws.String(cr.Status.AtProvider.CertificateAuthorityArn),
+				Principal:               aws.String("acm.amazonaws.com"),
+			}).Send(ctx)
+
+		} else {
+			_, err = e.client.DeletePermissionRequest(&awsacmpca.DeletePermissionInput{
+				CertificateAuthorityArn: aws.String(cr.Status.AtProvider.CertificateAuthorityArn),
+				Principal:               aws.String("acm.amazonaws.com"),
+			}).Send(ctx)
+		}
+
+		if err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, errPermissionFailed)
+		}
+	}
+	return managed.ExternalUpdate{}, nil
+}
+
+func (e *external) Delete(ctx context.Context, mgd resource.Managed) error {
+	cr, ok := mgd.(*v1alpha1.CertificateAuthority)
+	if !ok {
+		return errors.New(errUnexpectedObject)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Deleting())
+
+	response, err := e.client.DescribeCertificateAuthorityRequest(&awsacmpca.DescribeCertificateAuthorityInput{
+		CertificateAuthorityArn: aws.String(cr.Status.AtProvider.CertificateAuthorityArn),
+	}).Send(ctx)
+
+	if err != nil {
+		return errors.Wrap(err, errDelete)
+	}
+
+	if response != nil {
+		if response.CertificateAuthority.Status != awsacmpca.CertificateAuthorityStatusPendingCertificate {
+			_, err = e.client.UpdateCertificateAuthorityRequest(&awsacmpca.UpdateCertificateAuthorityInput{
```

Comment:
Is it the right behavior to attempt to update the CA if a delete was requested and the status is not pending? It seems like we will never reach the delete below.

Reply:
A PCA can be deleted if it is in the PENDING_CERTIFICATE, CREATING, EXPIRED, DISABLED, or FAILED state. In order to delete a CA in the ACTIVE state, we must first disable it, or else the delete request results in an exception.
`  delete failed: failed to delete the ACMPCA resource: InvalidStateException: The certificate authority is not in a valid state for deletion.`
  So added a check for PCA status and updating if required.